com.playeveryware.eos/Runtime/Core/EOS_SDK_Additions/Extensions/InputStateButtonFlagsExtensions.cs

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2024 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#if !EOS_DISABLE
+
+namespace PlayEveryWare.EpicOnlineServices.Extensions
+{
+    using Utility;
+    using System;
+    using System.Collections.Generic;
+    using Epic.OnlineServices.UI;
+
+    /// <summary>
+    /// Provides a means of parsing string representations of enum values for
+    /// the enum type InputStateButtonFlags.
+    /// </summary>
+    public static class InputStateButtonFlagsExtensions
+    {
+        /// <summary>
+        /// Returns the internal custom mappings.
+        /// </summary>
+        public static Dictionary<string, InputStateButtonFlags> CustomMappings { get; } = new()
+        {
+            { "DPadDown", InputStateButtonFlags.DPadDown },
+            { "DPadLeft", InputStateButtonFlags.DPadLeft },
+            { "DPadRight", InputStateButtonFlags.DPadRight },
+            { "DPadUp", InputStateButtonFlags.DPadUp },
+            { "FaceButtonBottom", InputStateButtonFlags.FaceButtonBottom },
+            { "FaceButtonLeft", InputStateButtonFlags.FaceButtonLeft },
+            { "FaceButtonRight", InputStateButtonFlags.FaceButtonRight },
+            { "FaceButtonTop", InputStateButtonFlags.FaceButtonTop },
+            { "LeftShoulder", InputStateButtonFlags.LeftShoulder },
+            { "LeftThumbstick", InputStateButtonFlags.LeftThumbstick },
+            { "LeftTrigger", InputStateButtonFlags.LeftTrigger },
+            { "RightShoulder", InputStateButtonFlags.RightShoulder },
+            { "RightThumbstick", InputStateButtonFlags.RightThumbstick },
+            { "RightTrigger", InputStateButtonFlags.RightTrigger },
+            { "SpecialLeft", InputStateButtonFlags.SpecialLeft },
+            { "SpecialRight", InputStateButtonFlags.SpecialRight }
+        };
```

Comment:
Because there is a 1:1 mapping between the `string` key for this `Dictionary` and the field name it maps to, there is actually no need for this `CustomMappings` field. 

The `CustomMappings` field member defined in other extension classes (like [`AuthScopeFlagsExtensions.cs`](https://github.com/PlayEveryWare/eos_plugin_for_unity/blob/2a18ec7e3fddff1c35baf37195a9d7989be1c4bd/com.playeveryware.eos/Runtime/Core/Utility/Extensions/AuthScopeFlagsExtensions.cs) is used because there are string values that need to be correlated to enum values whose names were changed at some point. 

Adding the `CustomMappings` field in these extension classes is done to support backwards-compatibility for users who have string values for their enum selections that no longer map to the current implementation of `AuthScopeFlags`. Since these values can be determined by calling `.ToString()` on the `InputStateButtonFlags` enum value this process is unnecessary.

Reply:
I wasn't sure that was the behaviour, so I made a tiny dotnetfiddle.

```csharp
[Flags]
	public enum FlaggedEnum : int
	{
		None = 0,
		Aaa = 1 << 0,
		Bbb = 1 << 1,
		Ccc = 1 << 2
	}
	
	public static void Main()
	{
		FlaggedEnum myFlaggedEnum = FlaggedEnum.Aaa;
		Console.WriteLine(myFlaggedEnum.ToString());
	}
```

The ToString resulted in printing out the name of the enum, so we can use that. I was concerned it would print out the numeric value.